loader.js

```diff
@@ -8,9 +8,13 @@ var NodeTargetPlugin = require("webpack/lib/node/NodeTargetPlugin");
 var LibraryTemplatePlugin = require("webpack/lib/LibraryTemplatePlugin");
 var SingleEntryPlugin = require("webpack/lib/SingleEntryPlugin");
 var LimitChunkCountPlugin = require("webpack/lib/optimize/LimitChunkCountPlugin");
+
+var extractTextPluginHmrRuntime = require.resolve("./hotModuleReplacement.js");
+
 module.exports = function(source) {
 	if(this.cacheable) this.cacheable();
-	return source;
+	// Even though this gets overwritten if extract+remove are true, without it, the runtime doesn't get added to the chunk
+	return "if(module.hot){require('" + extractTextPluginHmrRuntime + "');}\n" + source;
```

Comment:
Changed base to `master`.

Changed to template strings. Is the code style okay? I figured real newlines in the code were better than `\n`, but if I make the indentation match the surrounding code it'll appear with all those extra tabs in the built file itself.